traffic_ops/client/session.go

```diff
@@ -142,6 +156,29 @@ func (to *Session) login() (net.Addr, error) {
 	return remoteAddr, nil
 }
 
+func (to *Session) loginWithToken(token []byte) (net.Addr, error) {
+	path := apiBase + "/user/login/token"
+	resp, remoteAddr, err := to.rawRequest(http.MethodPost, path, token)
+	resp, remoteAddr, err = to.ErrUnlessOK(resp, remoteAddr, err, path)
+	if err != nil {
+		return remoteAddr, fmt.Errorf("requesting: %v", err)
+	}
+	defer resp.Body.Close()
+
+	var alerts tc.Alerts
+	if err := json.NewDecoder(resp.Body).Decode(&alerts); err != nil {
+		return remoteAddr, fmt.Errorf("decoding response JSON: %v", err)
+	}
+
+	for _, alert := range alerts.Alerts {
```

Comment:
would checking 2xx be better then string match on the alerts?

Reply:
Possibly. This is a copy of the logic from `loginWithAgent`, and I think a change to that logic should be global so that logging in always checks for success the same way.